lib/iris/fileformats/netcdf.py

```diff
@@ -741,6 +741,26 @@ def _setncattr(variable, name, attribute):
     return variable.setncattr(name, attribute)
 
 
+class _FillValueMaskCheckAndStoreTarget(object):
+    """
+    To be used with da.store. Remembers whether any element was equal to a
+    given value and whether it was masked, before passing the chunk to the
+    given target.
+
+    """
+    def __init__(self, target, fill_value=None):
+        self.target = target
+        self.fill_value = fill_value
+        self.contains_value = False
+        self.is_masked = False
+
+    def __setitem__(self, keys, arr):
+        if self.fill_value is not None:
+            self.contains_value = self.contains_value or self.fill_value in arr
+        self.is_masked = self.is_masked or ma.isMaskedArray(arr)
```

Comment:
I think this should be using "ma.is_masked" instead of "ma.isMaskedArray".
As stated [here](https://github.com/SciTools/iris/issues/2748#issuecomment-324578818) "make no difference between masked + non-masked arrays"